src/librbd/cache/pwl/rwl/PmemManager.h

```diff
@@ -0,0 +1,60 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_CACHE_RWL_PMEMMANAGER_H
+#define CEPH_LIBRBD_CACHE_RWL_PMEMMANAGER_H
+
+#include <libpmem.h>
+#include "librbd/cache/pwl/Types.h"
+namespace librbd {
+namespace cache {
+namespace pwl {
+namespace rwl {
+
+struct PmemDev {
+ public:
+  static PmemDev *pmem_create_dev(const char *path, const uint64_t size,
```

Comment:
instead of return a raw pointer in the public interface, i'd suggest returning a smart pointer, in this case, `unique_ptr<PmemDev>`.